test/crv-staking/fee-distributor.ts

```diff
@@ -0,0 +1,142 @@
+      sa.fundManager.address
+    );
+
+    await mta.deployed();
+
+    await mta.connect(sa.fundManager.signer).setTransfersAllowed(true);
+
+    // Get redeemer contract
+    Redeemer = await ethers.getContractFactory("Redeemer");
+    redeemer = await Redeemer.deploy(sa.fundManager.address, 1);
+
+    await redeemer.deployed();
+
+    IncentivisedVotingLockup = await ethers.getContractFactory(
+      "IncentivisedVotingLockup"
+    );
+    votingLockup = await IncentivisedVotingLockup.deploy(
+      mta.address,
+      redeemer.address,
+      await mta.name(),
+      await mta.symbol()
+    );
+
+    await votingLockup.deployed();
+
+    start = await getTimestamp();
+    FeeDistributor = await ethers.getContractFactory("FeeDistributor");
+    feeDistributor = await FeeDistributor.deploy(
+      votingLockup.address,
+      start,
+      mta.address,
+      sa.fundManager.address,
+      sa.fundManager.address
+    );
+
+    await feeDistributor.deployed();
+  });
+
+  describe("checking public variables", () => {
+    it("returns voting escrow contract", async () => {
+      expect(await feeDistributor.voting_escrow()).eq(votingLockup.address);
+    });
+    it("returns start time for distribution", async () => {
+      expect(await feeDistributor.start_time()).eq(
+        start.div(ONE_WEEK).mul(ONE_WEEK)
+      );
+    });
+    it("returns fee token", async () => {
+      expect(await feeDistributor.token()).eq(mta.address);
+    });
+    it("returns admin", async () => {
+      expect(await feeDistributor.admin()).eq(sa.fundManager.address);
+    });
+    it("returns emergency return address", async () => {
+      expect(await feeDistributor.emergency_return()).eq(
+        sa.fundManager.address
+      );
+    });
+  });
+
+  describe("burns", () => {
+    it("it reverts when not RBN token", async () => {
+      await expect(
+        feeDistributor.connect(sa.fundManager.signer).burn(votingLockup.address)
+      ).to.be.reverted;
+    });
+
+    it("it transfers RBN token to fee distributor", async () => {
+      let totalFundManagerRBNBalanceBefore = await mta.balanceOf(
+        sa.fundManager.address
+      );
+      let totalFeeDistributorRBNBalanceBefore = await mta.balanceOf(
+        feeDistributor.address
+      );
+
+      await mta
+        .connect(sa.fundManager.signer)
+        .approve(feeDistributor.address, totalFundManagerRBNBalanceBefore);
+
+      await feeDistributor.connect(sa.fundManager.signer).burn(mta.address);
+
+      let totalFundManagerRBNBalanceAfter = await mta.balanceOf(
+        sa.fundManager.address
+      );
+      let totalFeeDistributorRBNBalanceAfter = await mta.balanceOf(
+        feeDistributor.address
+      );
+
+      expect(totalFundManagerRBNBalanceAfter).eq(0);
+      expect(totalFeeDistributorRBNBalanceAfter).eq(
+        totalFeeDistributorRBNBalanceBefore.add(
+          totalFundManagerRBNBalanceBefore
+        )
+      );
+    });
+  });
+});
```

Comment:
what other tests you want here?

crv python tests for this code: [#1](https://github.com/curvefi/curve-dao-contracts/blob/master/tests/integration/FeeDistributor/test_checkpoint_ts.py) and [#2](https://github.com/curvefi/curve-dao-contracts/blob/master/tests/integration/FeeDistributor/test_distribute_fees_stateful.py)

the code is almost identical to original vyper [FeeDistributor.vy ](https://github.com/curvefi/curve-dao-contracts/blob/master/contracts/FeeDistributor.vy) w exception of renaming some methods to camel case in interface.